Source/Core/Core/HW/WiimoteEmu/Speaker.cpp

```diff
@@ -106,18 +106,17 @@ void Wiimote::SpeakerData(wm_speaker_data* sd)
 		return;
 	}
 
-	// Speaker Pan
-	unsigned int vol = (unsigned int)(m_options->settings[4]->GetValue() * 100);
-
+	// Speaker Pan using Constant Power Pan Law
+	double pan = m_options->settings[4]->GetValue() / 1.27;
+	double pan_prime = 3.14159265358979323846 * (pan + 1) / 4.0;
```

Comment:
Just as an explanation for the template approach:

My reasoning behind the template approach is it allows you to specify defined mathematical constants for types where it makes sense to actually have them, and fail on types where it clearly indicates a logic error (e.g. PI for int or any other integer type is sketchy usage at best), so it enforces correctness. It also has the benefit of allowing the use of those floating-point constants in constexpr expressions for calculations if need be (the define variant wouldn't be constexpr).